x/mint/keeper/abci.go

```diff
@@ -20,76 +18,12 @@ func (k Keeper) BeginBlocker(ctx context.Context, ic types.InflationCalculationF
 		return err
 	}
 
-	params, err := k.Params.Get(ctx)
+	// we pass -1 as epoch number to indicate that this is not an epoch minting,
+	// but a regular block minting. Same with epoch id "block".
+	err = mintFn(ctx, k.Environment, &minter, "block", -1)
```

Comment:
Is atomic behaviour ensured? The branch service is not used afaik

Reply:
In this point here, it would halt the chain. I think it's expected as it was the previous behavior. In case of the new epoched minting, that gets executed in a branch so it won't cause the chain to halt.